tabular/src/autogluon/tabular/learner/default_learner.py

```diff
@@ -110,7 +110,10 @@ def _fit(self, X: DataFrame, X_val: DataFrame = None, X_unlabeled: DataFrame = N
     def general_data_processing(self, X: DataFrame, X_val: DataFrame, X_unlabeled: DataFrame, holdout_frac: float, num_bag_folds: int):
         """ General data processing steps used for all models. """
         X = copy.deepcopy(X)
-
+        str_cols = list(X.select_dtypes('string').columns)
+        for col in str_cols:
+            logger.warning(f'Detected string dtype for column {col}, converting to object dtype')
+            X[col] = X[col].astype('object')
```

Comment:
All logic altering feature types must be located in the feature generator. Otherwise, someone could fit a feature generator outside of Predictor and pass it in to predictor.fit, which would force the type back to string.